tests/api/ClientTest.php

```diff
@@ -0,0 +1,282 @@
+        ]);
+        $response = $httpClient->request('DELETE', $iri);
+        $this->assertResponseIsSuccessful();
+        $response = $httpClient->request('GET', $iri);
+        $this->assertResponseStatusCodeSame(404);
+    }
+
+    public function testDeleteClientNotFound(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('DELETE', ClientMother::getNonExistentIri());
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.ClientMother::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClients(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
+
+    public function testGetClientsUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetNonExistentClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', ClientMother::getNonExistentIri());
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.ClientMother::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $client = ClientMother::base();
+        $client = $this->postClient($httpClient, $client);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => $client->getName()
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($client->getData());
+        $this->assertJsonContains($client->getContext());
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $client = ClientMother::base();
+        $client = $this->postClient($httpClient, $client);
+        $postScript = $this->getScriptId($httpClient, 'script_client_post');
+        $preScript = $this->getScriptId($httpClient, 'script_client_pre');
+        $updateClient = ClientMother::withAllParameters(
+            1,
+            "description updated",
+            true,
+            5,
+            [$postScript],
+            [$preScript],
```

Comment:
More than one script per array in both cases please to at least make sure that the handling of the array is correct.